src/service.js

```diff
@@ -182,12 +183,23 @@ class Service {
       data[this.id] = id;
     }
 
-    return this.Model
-      .update(params.query, { $set: data }, options)
-      .lean(this.lean)
-      .exec()
-      .then(() => this._getOrFind(id, params))
-      .catch(errorHandler);
+    let promise;
+
+    // We need this shitty hack because update doesn't return
+    // a promise properly when runValidators is true. WTF!
+    try {
+      promise = this.Model
```

Comment:
We can probably just `return` in both sections of the `try/catch`. Also, `.update` still does a partial update? I thought we have to use `$set`? If this is the exact same as `update` except with `multi: false` why don't we use that instead?